### PR TITLE
Add adaptive auth script for groups

### PR DESCRIPTION
**Pull request by somindatommy**
Related to https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/62

### Proposed changes in this pull request
Add adaptive auth script for groups

### CODE REVIEW

File: features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
```
Comment (mefarazath) [inline features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json:1]: @gomathyk can you help to review the text shown in the adaptive template UI?

File: features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "category": "user_based",
+  "name": "Group-Based",
+  "title": "Group-Based 2FA Template",
+  "summary": "Prompts 2FA to the users who belongs to any of the given set of groups.",
```
Comment (gomathyk) [inline features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json:5]:
```suggestion
  "summary": "Prompts 2FA to the users who belong to any of the given sets of groups.",
```

File: features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  "preRequisites": [
+    "Change the groupList parameter to an array of groups of which users need to enforce 2FA.",
```
Comment (gomathyk) [inline features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json:7]:
```suggestion
    "Change the groupList parameter to an array of groups for which users need to enforce 2FA.",
```

File: features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    "Modify the 1st and 2nd factor authentication option(s) from defaults as required."
+  ],
+  "parametersDescription": {
+    "groupsToStepUp": "An array of groups of which users need to enforce 2FA."
```
Comment (gomathyk) [inline features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json:11]:
```suggestion
    "groupsToStepUp": "An array of groups for which users need to enforce 2FA."
```

File: features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        "totp",
+        "FIDOAuthenticator"
+      ],
+      "federated": []
+    }
+  },
+  "code": [
+    "// This script will step up authentication for any user belong",
+    "// to one of the given groups.",
+    "// If the user is a member in the following groups, authentication will be stepped up",
```
Comment (gomathyk) [inline features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json:36]:
```suggestion
    "// If the user is a member of the following groups, authentication will be stepped up",
```

File: features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    "2": {
+      "local": [
+        "totp",
+        "FIDOAuthenticator"
+      ],
+      "federated": []
+    }
+  },
+  "code": [
+    "// This script will step up authentication for any user belong",
```
Comment (gomathyk) [inline features/application-mgt/org.wso2.carbon.identity.application.mgt.server.feature/resources/identity/authenticationTemplates/group.json:34]:
```suggestion
    "// This script will step up authentication for any user who belongs",
```

**Review by gomathyk — CHANGES_REQUESTED** [review]

**Comment by mefarazath** [comment]
@somindatommy let's name the file to something like group-based-2fa-step-up or something that helps to differentiate the template from others in future.

**Review by gomathyk — APPROVED** [review]

**Review by mefarazath — APPROVED** [review]

**Pull request merged**